conversation/frontend/src/hooks/useMessageReplyOrTransfer.tsx

```diff
@@ -0,0 +1,118 @@
+import { useDate, useEdificeClient } from '@edifice.io/react';
+import { Message, Recipients } from '~/models';
+import { DEFAULT_MESSAGE, useMessage } from '~/services';
+import { useI18n } from './useI18n';
+
+export interface MessageReplyOrTransferProps {
+  messageId: string | undefined;
+  replyMessageId?: string | null;
+  replyAllMessageId?: string | null;
+  transferMessageId?: string | null;
+}
+
+export function useMessageReplyOrTransfer({
+  messageId,
+  replyMessageId,
+  replyAllMessageId,
+  transferMessageId,
+}: MessageReplyOrTransferProps) {
+  const { data: messageOrigin } = useMessage({
+    messageId:
+      messageId ||
+      replyMessageId ||
+      replyAllMessageId ||
+      transferMessageId ||
+      '',
+  });
+  const { currentLanguage, user, userProfile } = useEdificeClient();
+  const { t } = useI18n();
+  const { formatDate } = useDate();
+  const displayRecipient = (recipients: Recipients) => {
+    const usersDisplayName = recipients.users
+      .map((user) => user.displayName)
+      .join(', ');
+    const groupsDisplayName = recipients.groups
+      .map((group) => group.displayName)
+      .join(', ');
+    return (
+      usersDisplayName +
+      (recipients.users.length > 0 && recipients.groups.length > 0
+        ? ', '
+        : '') +
+      groupsDisplayName
+    );
+  };
+
+  if (
+    messageOrigin?.id &&
+    (!!replyMessageId || !!replyAllMessageId || !!transferMessageId)
+  ) {
+    const messageTmp: Message = {
+      ...DEFAULT_MESSAGE,
+      language: currentLanguage,
+    };
+
+    const body = `<p></p><p></p>
+          <div class="${transferMessageId ? '' : 'conversation-history'}">
+              ${transferMessageId ? '<p><span style="font-size: 14px; font-weight:400;">' + t('transfer.title') + '</span></p>' : ''}
+              <p><span style="font-size: 14px; font-weight:400;">${t('transfer.from') + messageOrigin.from?.displayName}</span></p>
+              <p><span style="font-size: 14px; font-weight:400;">${t('transfer.date') + (!messageOrigin.date ? '' : formatDate(messageOrigin.date, 'long'))}</span></p>
+              <p><span style="font-size: 14px; font-weight:400;">${t('transfer.subject') + messageOrigin.subject}</span></p>
+              <p><span style="font-size: 14px; font-weight:400;">${t('transfer.to') + displayRecipient(messageOrigin.to)}</span></p>
+              ${messageOrigin.cc.users.length || messageOrigin.cc.groups.length ? '<p><span style="font-size: 14px; font-weight:400;">' + t('transfer.cc') + displayRecipient(messageOrigin.cc) + '</span></p>' : ''}
+            <p class="${transferMessageId ? '' : 'conversation-history-body'}">
+              ${messageOrigin.body}
+            </p>
+          </div>`;
+    messageTmp.body = body;
+
+    if (replyMessageId) {
```

Comment:
si le typage change : 
```suggestion
    switch (action) {
      case "reply" : ....
```

et 1 seul return en fin de hook.